### DISCUSSION TITLE
.Goo format, 'slow' retract speed is being ignored if UVtools is used

### DISCUSSION THREAD

**Discussion by bryanseger** [sn4k3/UVtools#743] [General]
Using the latest 3.15.1, we are seeing the 'slow' retract speed is simply being ignored on an Elegoo Mars 4 DLP if UVtools is used. Attached are two seemingly identical files, but one is straight from Tango and the other is Tango-->UVtools. The Tango file works as expected, but the Tango-UVtools file does not (in that the slow retract speed is ignored, but everything works).

Both files seem the same from a settings standpoint, but do you see a difference in the files or a reason why we're seeing this?
[Cone without UV Tools.zip](https://github.com/sn4k3/UVtools/files/12221371/Cone.without.UV.Tools.zip)

**Comment by sn4k3** [comment]
You applied the Wait time suggestion in UVtools which make file to follow layer data instead of global properties.
Still the layer properties include the TSMC lift and retract correctly. If only retract doesn't work it can be another bug on Goo printer firmware that is ignoring tsmc retracts that way.

**Comment by msspad** [comment]
Hey Bryan, do you know whether setting different WaitTimeBeforeCure for different layers works with Mars 4 DLP?

I'm thinking of buying the Mars 4 Ultra and individual WaitTimeBeforeCure are really important for my prints. At least the LayerDef class in UVTool's GooFile.cs has the property WaitTimeBeforeCure but I don't know about the Elegoo Mars firmware...

Would be really nice to give a hint, thanks
